tests/drivers/gpio/gpio_ite_it8xxx2_v2/src/main.c

```diff
@@ -0,0 +1,462 @@
+	zexpect_equal(reg_gpotr, 0, "reg_gpotr=%x", reg_gpotr);
+	zexpect_equal(reg_p18scr, 0);
+	zexpect_equal(reg_gpcr[TEST_PIN], GPCR_PORT_PIN_MODE_INPUT, "reg_gpcr[%d]=%x", TEST_PIN,
+		      reg_gpcr[TEST_PIN]);
+	reg_gpdmr = (uint8_t)~BIT(TEST_PIN);
+	zassert_false(gpio_pin_get(gpio_dev, 1));
+	reg_gpdmr = BIT(TEST_PIN);
+	zassert_true(gpio_pin_get(gpio_dev, 1));
+}
+
+ZTEST(gpio_ite_it8xxx2_v2, test_get_active_low)
+{
+	zassert_true(device_is_ready(gpio_dev));
+	zassert_ok(gpio_pin_configure(gpio_dev, TEST_PIN, GPIO_INPUT | GPIO_ACTIVE_LOW));
+	zexpect_equal(reg_gpotr, 0, "reg_gpotr=%x", reg_gpotr);
+	zexpect_equal(reg_p18scr, 0);
+	zexpect_equal(reg_gpcr[TEST_PIN], GPCR_PORT_PIN_MODE_INPUT, "reg_gpcr[%d]=%x", TEST_PIN,
+		      reg_gpcr[TEST_PIN]);
+	reg_gpdmr = (uint8_t)~BIT(TEST_PIN);
+	zassert_true(gpio_pin_get(gpio_dev, 1));
+	reg_gpdmr = BIT(TEST_PIN);
+	zassert_false(gpio_pin_get(gpio_dev, 1));
+}
+
+ZTEST(gpio_ite_it8xxx2_v2, test_interrupt_edge_rising)
+{
+	zassert_true(device_is_ready(gpio_dev));
+	zassert_ok(gpio_pin_configure(gpio_dev, TEST_PIN, GPIO_INPUT | GPIO_ACTIVE_HIGH));
+
+	gpio_init_callback(&callback_struct, &callback, BIT(TEST_PIN));
+	zassert_ok(gpio_add_callback(gpio_dev, &callback_struct));
+	zassert_ok(gpio_pin_interrupt_configure(gpio_dev, TEST_PIN, GPIO_INT_EDGE_TO_ACTIVE));
+	zexpect_equal(reg_gpotr, 0, "reg_gpotr=%x", reg_gpotr);
+	zexpect_equal(reg_p18scr, 0);
+	zexpect_equal(reg_gpcr[TEST_PIN], GPCR_PORT_PIN_MODE_INPUT, "reg_gpcr[%d]=%x", TEST_PIN,
+		      reg_gpcr[TEST_PIN]);
+	zexpect_equal(reg_wubemr, 0, "reg_wubemr=%x", reg_wubemr);
+	zexpect_equal(reg_wuemr, 0, "reg_wuemr=%x", reg_wuemr);
+	zexpect_equal(reg_wuesr, TEST_MASK, "reg_wuesr=%x", reg_wuesr);
+	reg_wuesr = 0;
+
+	reg_gpdmr = BIT(TEST_PIN);
+	/* Mock the hardware interrupt. */
+	posix_sw_set_pending_IRQ(TEST_IRQ);
+	k_sleep(K_MSEC(100));
+	zassert_equal(callback_called, 1, "callback_called=%d", callback_called);
+}
+
+ZTEST(gpio_ite_it8xxx2_v2, test_interrupt_enable_disable)
+{
+	zassert_true(device_is_ready(gpio_dev));
+	zassert_ok(gpio_pin_configure(gpio_dev, TEST_PIN, GPIO_INPUT | GPIO_ACTIVE_HIGH));
+
+	gpio_init_callback(&callback_struct, &callback, BIT(TEST_PIN));
+	zassert_ok(gpio_add_callback(gpio_dev, &callback_struct));
+	zassert_ok(gpio_pin_interrupt_configure(gpio_dev, TEST_PIN, GPIO_INT_EDGE_TO_ACTIVE));
+	zexpect_equal(reg_gpotr, 0, "reg_gpotr=%x", reg_gpotr);
+	zexpect_equal(reg_p18scr, 0);
+	zexpect_equal(reg_gpcr[TEST_PIN], GPCR_PORT_PIN_MODE_INPUT, "reg_gpcr[%d]=%x", TEST_PIN,
+		      reg_gpcr[TEST_PIN]);
+	zexpect_equal(reg_wubemr, 0, "reg_wubemr=%x", reg_wubemr);
+	zexpect_equal(reg_wuemr, 0, "reg_wuemr=%x", reg_wuemr);
+	zexpect_equal(reg_wuesr, TEST_MASK, "reg_wuesr=%x", reg_wuesr);
+	reg_wuesr = 0;
+
+	reg_gpdmr = BIT(TEST_PIN);
+	/* Mock the hardware interrupt. */
+	posix_sw_set_pending_IRQ(TEST_IRQ);
+	k_sleep(K_MSEC(100));
+	zassert_equal(callback_called, 1, "callback_called=%d", callback_called);
+
```

Comment:
For consistency, clear `gpdmr` between the sub-tests so simulate the low to high transition.

Reply:
done